content/en/blog/prerendering-preset-vite.md

```diff
@@ -0,0 +1,348 @@
+	- In fact, that's how we build the very page you're reading! `fetch('/content/blog/preact-prerender.json')`, which is what triggers when you navigate to this page, (roughly) translates to `new Response(await fs.readFile('/content/blog/preact-prerender.json'))` during prerendering. We read the file, wrap it in a `Response` to mimic a network request, and supply it back to your app -- your app can use the same `fetch()` request during prerendering and on the client.
+	- Pairing this with suspense and an async SSR implementation provides a really great DX.
+- Crawling Links
+	- Partly supported by the user-provided `prerender()` function export, partly by the plugin, you can return a set of links upon prerendering the page (`preact-iso` makes this wonderfully simple) which will be added to the plugin's list of URLs to prerender. This will allow the plugin to crawl your site at build time, finding more and more pages to prerender naturally.
+	- You can also provide links manually via the plugin options or by appending some to those that `preact-iso` returns, as we show above in the Full API Example. This is especially useful for error pages, like a `/404`, that might not be linked to but you still want to have it prerendered.
+
+...and perhaps the biggest advantage:
+
+- Toggle it by flipping a Boolean in your config file
+	- Because we're not a wrapper, and because you don't need to alter your source code in order to support it beyond some window checks, there's no lock-in whatsoever. If you decide to move away, or you want to do some testing on your output, all you need to do is flip a Boolean and you're back to a plain SPA with Vite.
+	- As we've mentioned a few times, prerendering is at its best when it is as near to "drop-in" as possible and that includes being able to drop back out on a whim. It's important to us that you can go from an SPA to prerendering and back again with minimal effort.
+
+## Final Notes
+
+The Vite team would probably like us to mention that this plugin does introduce a tiny patch to the generated client code, and that they (the Vite team) don't necessarily endorse running the browser bundles in Node.
+
+The patch in question is as follows:
+
+```diff
+// src/node/plugins/importAnalysisBuild.ts
+-if (__VITE_IS_MODERN__ && deps && deps.length > 0) {,
++if (__VITE_IS_MODERN__ && deps && deps.length > 0 && typeof window !== 'undefined') {,
+	 const links = document.getElementsByTagName('link')
+	 ...
+```
+
+As attempting to execute `document.getElementsByTagName` will error in Node where there is no `document` we simply add an additional condition to the preloader so that it makes no attempt to run in Node and that's it. Just the partial change of this one line.
```

Comment:
```suggestion
As attempting to execute `document.getElementsByTagName` will error in Node where there is no `document`, we simply add an additional condition to the preloader so that it makes no attempt to run in Node, and that's it. Just the partial change of this one line.
```